firmware/ventilator-controller-stm32/Core/Test/Pufferfish/Protocols/messages.txt

```diff
@@ -0,0 +1,181 @@
+      THEN("After the write method is called, the output buffer remains unchanged")
+
+    WHEN("The output buffer is too small to hold the generated body")
+      THEN("The write method reports invalid length status")
+      THEN("After the write method is called, the output buffer remains unchanged")
+
+    WHEN("The size of the output buffer is zero")
+      THEN("The write method reports invalid length status")
+
+    // Intent: To show that the write method does not change the internal "type" member of the message class
+    WHEN("The internal type member is set to an arbitrary value and then write method is called")
+    // setup:
+    // type is set to 4
+    // A sensor measurements payload message is set, which has value of type as 2
+      THEN("The write method reports ok status")
+      THEN("After the write method is called, type member remains unchanged)
+
+
+  GIVEN("A Message object constructed with StateSegment Taggedunion, default message types and a max size of 126 bytes")
+    WHEN("The payload is a sensor measurements message and write method is called")
+      THEN("The write method reports ok status")
+      THEN("The body's header correctly stores the value of message type obtained from the message payload tag")
+      THEN("The output buffer is as expected")
+
+  GIVEN("A Message object constructed with StateSegment Taggedunion, default message types and a max size of 508 bytes")
+    WHEN("The sensor measurments message data is written")
+      THEN("The write method reports ok status")
+      THEN("The body's header correctly stores the value of message type obtained from the message payload tag")
+      THEN("The output buffer is as expected")
+
+/* Write function for each of the message types (example: sensor_measurements)
+    WHEN("The payload is a sensor measurements message and write method is called")
+    // setup: sensor_measurements.flow = 50
+      THEN("The write method reports ok status")
+      THEN("The body's header correctly stores the value of message type obtained from the message payload tag")
+      THEN("The output buffer is as expected")
+/*
+
+/* Parse function
+
+Scenario: Message: The parse method correctly updates internal message type and sets the message payload fields after decoding from the input buffer
+
+  GIVEN("A Message object constructed with StateSegment Taggedunion, default message types and a max size of 252 bytes")
+
+    WHEN("A body with an encoded payload and 1 byte header of value equal to 0 is parsed")
+      THEN("The parse method reports invalid type status")
+      THEN("After the parse method is called, The value assigned to the type member is equal to the input_buffer body's header")
+      THEN("The payload Taggedunion tag is set to Unknown")
+      THEN("The message payload fields are not set")
+
+    WHEN("An empty input buffer body is parsed")
+      THEN("The parse method reports invalid length status")
+      THEN("The payload Taggedunion tag is set to Unknown")
```

Comment:
I don't think `parse` sets the internal members if it returns `invalid_length`, I think it just leaves them at whatever their previous values were. If this is accurate, we should design tests to check that it actually does leave them at the previous values (this will involve adding a new GIVEN with different initial values of the members, and maybe the GIVENs should be reorganized under a separate scenario)

Reply:
Did you find that parse sets the internal members, such as `tag`, even if the method returns `invalid_length`?

Reply:
No, it does not, it remains unchanged as the payload is uninitialized in this GIVEN, it is unknown before or after parse